src/metadata.rs

```diff
@@ -175,6 +181,16 @@ impl ModuleMetadata {
             .get(key)
             .ok_or(MetadataError::StorageNotFound(key))
     }
+
+    /// Get a constant's metadata by name
+    pub fn constant(
+        &self,
+        key: &'static str,
```

Comment:
```suggestion
        key: &str,
```
No need to restrict it to `'static`

Reply:
I see not some of the other similar methods are also `'static`, but I don't think they need to be if we make the respective `Error` variants have a `String` instead.